pkg/machine/applehv/machine.go

```diff
@@ -451,13 +513,81 @@ func (m *MacMachine) SSH(name string, opts machine.SSHOptions) error {
 	return machine.CommonSSH(username, m.IdentityPath, m.Name, m.Port, opts.Args)
 }
 
+// deleteIgnitionSocket retrieves the ignition socket, deletes it, and returns a
+// pointer to the `VMFile`
+func (m *MacMachine) deleteIgnitionSocket() (*machine.VMFile, error) {
+	ignitionSocket, err := m.getIgnitionSock()
+	if err != nil {
+		return nil, err
+	}
+	if err := ignitionSocket.Delete(); err != nil {
+		return nil, err
+	}
+	return ignitionSocket, nil
+}
+
+// getIgnitionVsockDeviceAsCLI retrieves the ignition vsock device and converts
+// it to a cmdline format
+func getIgnitionVsockDeviceAsCLI(ignitionSocket *machine.VMFile) ([]string, error) {
```

Comment:
This should be

func (ignitionSocket *machine.VMFile(getIgnitionVsockDeviceAsCLI

Reply:
So you would like this to be a method of `machine.VMFile`?

Reply:
Yes, that sounds good :+1:

Reply:
I personally think that making this a method of `machine.VMFile` would get a little messy since the code is `vfkit` specific. `getIgnitionVsockDevice` is an un-exported func in the `applehv` package which returns a `vfkit` specific type. I'm not sure it's a great idea to pull that code out of the `applehv` package.

Reply:
You can create methods that are not exported or declared in an interface, it should still work.